pkg/translator/loki/logs_to_loki.go

```diff
@@ -114,6 +114,73 @@ func LogsToLokiRequests(ld plog.Logs, defaultLabelsEnabled map[string]bool) map[
 	return requests
 }
 
+func LogsToLokiRequestsWithResourceFieldInJSONFormat(ld plog.Logs, defaultLabelsEnabled map[string]bool) map[string]PushRequest {
+	groups := map[string]pushRequestGroup{}
+
+	rls := ld.ResourceLogs()
+	for i := 0; i < rls.Len(); i++ {
+		ills := rls.At(i).ScopeLogs()
+		resource := rls.At(i).Resource()
+
+		for j := 0; j < ills.Len(); j++ {
+			logs := ills.At(j).LogRecords()
+			scope := ills.At(j).Scope()
+			for k := 0; k < logs.Len(); k++ {
+				log := logs.At(k)
+				tenant := GetTenantFromTenantHint(log.Attributes(), resource.Attributes())
+				group, ok := groups[tenant]
+				if !ok {
+					group = pushRequestGroup{
+						report:  &PushReport{},
+						streams: make(map[string]*push.Stream),
+					}
+					groups[tenant] = group
+				}
+
+				entry, err := LogToLokiEntryWithResourceFieldInJSONFormat(log, resource, scope, defaultLabelsEnabled)
```

Comment:
if this is the only line that differs this version from the current one, would it be worth splitting this function into a more generic one that received an encoder func (`LogToLokiEntryWithResourceFieldInJSONFormat` or the current one)? In case of a bug in this function between now and when the flag is removed, splitting the issue would make fixing it easier.

Reply:
I'm not quite sure I like that idea, as we'd be using the pipeline data to store business attributes of the component. Even if this is temporary, I think I would prefer the code duplication if those are the only two options.